.github/workflows/release-please.yml

```
@@ -0,0 +1,100 @@
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+name: release-please
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    outputs:
+      release_created: ${{ steps.release.outputs.release_created }}
+      tag_name: ${{ steps.release.outputs.tag_name }}
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+      - uses: googleapis/release-please-action@v4
+        id: release
+
+  build:
+    name: Build distribution
+    needs:
+      - release-please
+    if: ${{ needs.release-please.outputs.release_created }}  # only publish on release
+    runs-on: ubuntu-latest
```

If we change the default permission, [`actions: write`](https://docs.github.com/en/rest/actions/artifacts?apiVersion=2022-11-28) needs to be granted on this action.

```suggestion
    runs-on: ubuntu-latest
    permissions
      actions: write
```